### ISSUE TITLE
[Feature] MRP retry interval tuning for Thread

### ISSUE THREAD

**Issue by Damian-Nordic** [project-chip/connectedhomeip#33047]
### Feature description

**TLDR**
We hear from some members that the current MRP retry intervals for Thread devices are insufficient, resulting in too eager retransmissions and a positive feedback loop that exacerbates network congestion.

An overloaded Thread router that is an intermediate hop for many parallel conversations may become a bottleneck and then it may be burdened with even more traffic due to spurious retransmissions.

**Some history**
1. Initially, Matter used 300ms as default SAI/SII for all types of devices.
2. At the beginning of 2023, we [increased default SAI/SII for Thread devices to 800ms](https://github.com/project-chip/connectedhomeip/pull/24549). The choice of 800ms was made based on measurements done using the following line topology:
   
   ![image](https://github.com/project-chip/connectedhomeip/assets/66667989/2374ac99-048e-4195-9508-995940a8ba7a)

   This PR has also introduced `CHIP_CONFIG_MRP_RETRY_INTERVAL_SENDER_BOOST` configuration that allowed a Thread sender to add an extra time to the MRP backoff. This was necessary to address scenarios in which a Thread device sends a message to a Wi-Fi device that uses low SAI/SII (300ms/500ms):

   ![image](https://github.com/project-chip/connectedhomeip/assets/66667989/22e961d3-5343-4524-856c-f5eec741892c)

3. [Quick search](https://github.com/search?q=repo%3Aproject-chip%2Fconnectedhomeip%20CHIP_CONFIG_MRP_LOCAL_ACTIVE_RETRY_INTERVAL&type=code) reveals that other silicon vendors increased these parameters as well in the past.

**Proposal**
Let's change the default SAI/SII for Thread devices to 2s.

It might be said that this configuration is device-specific and doesn't have to be aligned. However, my concern is that if it's not unified to some degree, controller makers might compensate for too low values used by some accessories, which could lead to excessively long retry intervals in the case of other accessories. Therefore, I believe it's beneficial to align these settings to a certain degree to assure better interoperability.

There's also a question what to do with `CHIP_CONFIG_MRP_RETRY_INTERVAL_SENDER_BOOST`. The most simple answer would be to also increase it to, say, `1500ms`. But that would cause that two Thread devices talking to each other would use pretty long retry intervals: `MRP_backoff(2s) + 1.5s`.

As far as I can tell, there's currently no way to tell if a receiver is a Wi-Fi or Thread device and apply the sender boost only in the former case. Hence, it might be a good idea to replace `CHIP_CONFIG_MRP_RETRY_INTERVAL_SENDER_BOOST` with `CHIP_CONFIG_MRP_MIN_RETRY_INTERVAL` (configured to 2s for Thread devices) so that the retry interval is calculated as `max(MRP_backoff(peer.SAI or SII), CHIP_CONFIG_MRP_MIN_RETRY_INTERVAL)`.

Any other ideas to improve the MRP behavior without big spec changes?

### Platform

all

### Platform Version(s)

ToT
